centrallix/objectsystem/obj_datatypes.c

```diff
@@ -1617,9 +1636,9 @@ objDataCompare(int data_type_1, void* data_ptr_1, int data_type_2, void* data_pt
 		    case DATA_T_MONEY:
 		        objDataToMoney(DATA_T_STRING, data_ptr_1, &m_v);
 			m = (pMoneyType)data_ptr_2;
-			if (m_v.WholePart > m->WholePart) cmp_value = 1;
-			else if (m_v.WholePart < m->WholePart) cmp_value = -1;
-			else cmp_value = m_v.FractionPart - m->FractionPart;
+			if (m_v.MoneyValue/10000 > m->MoneyValue/10000) cmp_value = 1;
+			else if (m_v.MoneyValue/10000 < m->MoneyValue/10000) cmp_value = -1;
+			else cmp_value = m_v.MoneyValue%10000 - m->MoneyValue%10000;
```

Comment:
you should be able to just compare MoneyValue without separately comparing the whole/fraction sections

Reply:
and then you can just set cmp_value to 0 in the final else